src/lua/bindings/script.cpp

```diff
@@ -156,9 +157,9 @@ namespace lua::script
 	// Name: execute_as_script
 	// Param: script_name: string: target script thread.
 	// Param: func: function: function that will be executed once in the script thread.
-	static void execute_as_script(const std::string& script_name, sol::function func)
+	static void execute_as_script(const std::string& script_name, sol::protected_function func_, sol::this_state state)
```

Comment:
Change from func_ to something reasonable, like parameter_func or something.

Reply:
example code:
```
function run()
	log.info(SCRIPT.GET_THIS_SCRIPT_NAME())
end

script.run_in_fiber(function(state)
	script.execute_as_script("shop_controller", function()
		run()
	end)
	log.info("function")
	script.execute_as_script("shop_controller", run)
	log.info("function")
end)
```

Reply:
What does that have to do with what I asked?

Reply:
in some natives, it needs to be used here.

Reply:
I asked you to change the name of the parameter.